index.js

```diff
@@ -36,49 +36,48 @@ ext           = derive(msg_key, 'ext')
 header_box[32](mac_tag[16], header[16](offset[2],flags[1],hdr_ext[13])
 */
 
-exports.box = function (ptxt, external_nonce, msg_key, recipient_keys, ephemerals, padding) {
-
+exports.box = function (ptxt, external_nonce, msg_key, recipient_keys, ephemerals = [], padding) {
   if(!Array.isArray(ephemerals)) throw new Error('ephemerals must be an array')
-
   if(ephemerals.length) throw new Error('ephemerals not yet implemented, must be empty array')
 
   var msg_read_cap  = derive(msg_key, 'read_cap')
-  var hdr_key     = derive(msg_read_cap, 'header')
-
+    var hdr_key     = derive(msg_read_cap, 'header')
+    var body_key    = derive(msg_read_cap, 'body')
+      var box       = derive(body_key, 'box')
+      var box_nonce = derive(body_key, 'box_nonce', 192)
+      // mix: I'd be keen to rename some of these cos there are multiple "box"s here
 
   var header_length = (
     32
   + (32 * recipient_keys)
   + (ephemerals.length ? 32 * (ephemerals.length + 1) : 0)
   + padding ? padding.length : 0
   )
+  // mix: in the diagram header is only the first 32 bytes
+  // this is much more so should change diagram or change name
 
-  var length = (header_length + ptxt.length)
-
-  var buffer = Buffer.alloc(length)
+  var buffer = Buffer.alloc(header_length + ptxt.length)
   var header_box = buffer.slice(0, 32)
 
   var _header = header_box.slice(16)//Buffer.alloc(16)
   //header_length tells you where the body starts, it is encrypted into header_box.
   _header.writeUInt32LE(0, header_length)
 
-  na.crypto_box_easy(header_box, _header, hdr_key, nonce??) //XXX nonce??
+  na.crypto_secretbox_easy(header_box, _header, nonce??, hdr_key) //XXX nonce??
```

Comment:
This change right here makes me wish na used destructuring for arguments - the order change of where `hdr_key` and `nonce` are placed seems like an easy thing to mix up.

Reply:
Nope, this was just 2 typos on this starting gesture of code which needed changing (I think).
From docs  https://github.com/sodium-friends/sodium-native/tree/v2.3.0
```
crypto_box_easy(ciphertext, message, nonce, publicKey, secretKey)
crypto_secretbox_easy(ciphertext, message, nonce, secretKey)
```

so nonce should always be first. I get what you mean about destructuring, but I think the greater goal is to keep the same argument signatures as the lower level libsodium https://download.libsodium.org/doc/secret-key_cryptography/secretbox